src/components/Reports/GoalCalculator/GoalCard/GoalCard.test.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { AdapterLuxon } from '@mui/x-date-pickers/AdapterLuxon';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { DateTime } from 'luxon';
+import { GoalCard } from './GoalCard';
+
+const TestComponent: React.FC<React.ComponentProps<typeof GoalCard>> = (
+  props,
+) => (
+  <LocalizationProvider dateAdapter={AdapterLuxon}>
+    <GoalCard {...props} />
+  </LocalizationProvider>
+);
+
+describe('GoalCard', () => {
+  const defaultProps = {
+    goalId: 1,
+    goalTitle: 'Test Goal',
+    goalAmount: 1000,
+    goalDate: DateTime.fromISO('2024-06-01'),
+    starred: false,
+    onStarToggle: jest.fn(),
+    onDelete: jest.fn(),
+    onView: jest.fn(),
+  };
```

Comment:
Normally we define mock functions as separate variables when we need to test when they're called in tests. Then we can reference them as `onStarToggle` instead of `defaultProps.onStarToggle` when checking whether they're called. If you *don't* need to check whether they are called, there's no need to store them in a variable.

```suggestion
  const onStarToggle = jest.fn();
  const onDelete = jest.fn();
  const onView = jest.fn();
  const defaultProps = {
    goalId: 1,
    goalTitle: 'Test Goal',
    goalAmount: 1000,
    goalDate: DateTime.fromISO('2024-06-01'),
    starred: false,
    onStarToggle,
    onDelete,
    onView,
  };
```